### ISSUE TITLE
storage-resize-images: Resize Images Extension incorrectly names files containing "."

### ISSUE THREAD

**Issue by augustkimo** [firebase/extensions#476]
### [REQUIRED] Step 2: Describe your configuration

- Extension name: (`storage-resize-images`)
- Extension version: 0.1.12

### [REQUIRED] Step 3: Describe the problem
The resize image extension is incorrectly naming files which are named from a timestamp. The "." in the file name is likely the issue. 

Normally files from the extension are automatically named the same with the width/height following.
ex: profilePic  ->  profilePic_148x148

When renaming files which are named from a timestamp, the width/height goes into the middle of the name.
ex: 2020-10-22T15:29:00.736769  ->  2020-10-22T15:29:00_148x148.736769
The outputted file name should be: 2020-10-22T15:29:00.736769_148x148

#### Steps to reproduce:
1. Upload a photo using a time stamp (this bug might also occur when a "." is in the file name).
2. Have Firebase Resize Photo extension enabled to not delete the original file, and save the new resized image under a "thumbnails" directory.
3. New thumbnail image name will be named incorrectly. 

##### Expected result
A file named: 2020-10-22T15:29:00.736769_148x148

##### Actual result
A file named: 2020-10-22T15:29:00_148x148.736769

##### Suggested Fix / Problem
Edit: The problem is that when uploading a file to storage which doesn't contain a file extension in the set name (picture.of.cats versus picture.of.cats.jpg), `fileNameWithoutExtension` is still called. This results in the the text after the furthest period being detected as the extension, when it is really part of the file name.  Maybe there should be a check before `fileNameWithoutExtension` to ensure that a there is a valid extension in the file name first. 

~~I assume the extension looks for the first "." in the file name, and names the file without the file type extension. (picture.png would be named picture_148x148).~~

~~The code which names the new photo should look for the very last "." when renaming, not the first one. 
ex: myPicture.is.named.with.dots.png should -> myPicture.is.named.with.dots_148x148
but instead the extension will name the picture: myPicture_148x148.is.named.with.dots~~

**Comment by russellwheatley** [comment]
Hey @augustkimo, I'm struggling to recreate the problem you're encountering. Here's a screenshot using `storage-resize-images` version `0.1.12` which have the correct file names:
![Screenshot 2020-10-23 at 13 33 45](https://user-images.githubusercontent.com/16018629/97004098-69a78100-1534-11eb-956e-a870150bd4a5.png)

I've also extracted the implementation for creating a filename from the extension and ran it in isolation on Node 8, 10 & 12. It still returns the correct file name. 
```js
const path = require('path');
// imitate storage object passed into extension
const object = {
  name:'2020-10-22T15:29:00.736769.jpg'
}

const extractFileNameWithoutExtension = (
  filePath,
  ext
) => {
  return path.basename(filePath, ext);
};

const filePath = object.name; // File path in the bucket.
const fileExtension = path.extname(filePath);
const size = '200x200';
const fileNameWithoutExtension = extractFileNameWithoutExtension(
      filePath,
      fileExtension
    );

const resizedFileName = `${fileNameWithoutExtension}_${size}${fileExtension}`;    

console.log(resizedFileName);
```

Not sure how this is happening to your file names. If there are any other instructions for reproducing, I'd be happy to help.

**Comment by augustkimo** [comment]
Hi @russellwheatley, thanks for looking into this. Not a major bug as client side file name can easily be changed but thought I'd still report it. 

I am using Node v12.16.3

I think there are two difference between our uploads:
1. In storage my file names don't show the extension by default. I don't remember/know if this is a optional setting, but I don't remember setting this. (For you file name is: picture.jpg, while for me it's just named: picture)
2. Since what I said in point 1 I think `extractFileNameWithoutExtension` is running after Storage has already removed the extension from the file name.

![cats bug](https://user-images.githubusercontent.com/72067396/97008577-00fadd00-1511-11eb-8214-dd4f6310887a.PNG)

Here is the code I am using with Flutter with firebase_storage: ^3.1.6:
```
_uploadAdditionalFiles(uid, image) async {
    //String imageName = DateTime.now().toIso8601String();
    //imageName = imageName.replaceAll(".", "");

    String imageName = "cats.with.full.stops";
    StorageReference storageReference = FirebaseStorage.instance.ref().child('profiles/$uid/photos/additional/$imageName');
    StorageUploadTask uploadTask = storageReference.putFile(image);
    await uploadTask.onComplete;
    storageReference.getDownloadURL().then((fileURL) {
      setState(() {
        additionalPhotoUrls.add(fileURL);
      });
    });
  }
```

**Comment by jhuleatt** [comment]
Hey @russellwheatley, just to make sure I understand, is this marked as blocked because it is waiting on #479?

**Comment by russellwheatley** [comment]
@jhuleatt yes, I thought it was worth waiting until that PR and its increased test coverage was merged first. I'm not sure if that's the appropriate way to use the "blocked" column 😅

**Comment by jhuleatt** [comment]
Yep, that works, just making sure! I've merged #479. I've also added this to this week's milestone, but we can take it off if it isn't ready.
